qiskit/algorithms/eigensolvers/numpy_eigensolver.py

```diff
@@ -0,0 +1,297 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The NumPy eigensolver algorithm."""
+
+from __future__ import annotations
+
+from typing import Callable
+import logging
+import numpy as np
+from scipy import sparse as scisparse
+
+from qiskit.opflow import PauliSumOp
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+from qiskit.quantum_info import Statevector
+from qiskit.utils.validation import validate_min
+
+from ..exceptions import AlgorithmError
+from .eigensolver import Eigensolver, EigensolverResult
+from ..list_or_dict import ListOrDict
+
+logger = logging.getLogger(__name__)
+# pylint: disable=invalid-name
```

Comment:
Is this still needed - I see the old code has it - wondering what name is invalid as such

Reply:
haha I was also wondering and tried to remove it, but pylint doesn't like `self._k`

Reply:
Ah ok, that does seem to jog my memory!